source/patterns/@aws-solutions-constructs/aws-s3-sns/README.md

```diff
@@ -0,0 +1,109 @@
+# aws-s3-sns module
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> All classes are under active development and subject to non-backward compatible changes or removal in any
+> future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_s3_sns`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-s3-sns`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.s3sns`|
+
+## Overview
+This AWS Solutions Construct implements an Amazon S3 Bucket that is configured to send S3 event messages to an Amazon SNS topic.
+
+
+Here is a minimal deployable pattern definition:
+
+Typescript
+``` typescript
+import { Construct } from 'constructs';
+import { Stack, StackProps } from 'aws-cdk-lib';
+import { S3ToSns } from "@aws-solutions-constructs/aws-s3-sns";
+
+new S3ToSns(this, 'S3ToSNSPattern', {});
+```
+
+Python
+```python
+from aws_solutions_constructs.aws_s3_sns import S3ToSns
+from aws_cdk import Stack
+from constructs import Construct
+
+S3ToSns(self, 'S3ToSNSPattern')
+```
+
+Java
+``` java
+import software.constructs.Construct;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awsconstructs.services.s3sns.*;
+
+new S3ToSns(this, "S3ToSNSPattern", new S3ToSnsProps.Builder()
+        .build());
+```
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingBucketObj?|[`s3.Bucket`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.IBucket.html)|Existing instance of S3 Bucket object. If this is provided, then also providing bucketProps is an error. |
```

Comment:
Just to confirm - this needs a bucket object rather than an interface?

Reply:
I've just followed the standard existingBucketObj definition we have elsewhere, [here](https://github.com/awslabs/aws-solutions-constructs/blob/main/source/patterns/@aws-solutions-constructs/aws-lambda-s3/README.md?plain=1#L85) and [here](https://github.com/awslabs/aws-solutions-constructs/blob/main/source/patterns/@aws-solutions-constructs/aws-s3-sqs/README.md?plain=1#L65) for example

Reply:
OK for now, let's try both IBucket and Bucket while building to see which is required.

Reply:
The underlying helpers require Bucket. I've updated the README to match the expected type.